kvmagent/kvmagent/plugins/volume_backup_nbd_conversion.py

```
@@ -0,0 +1,727 @@
+        path = _write_forest_request(request)
+        try:
+            cmdstr = '%s -url %s forest %s -args-json-file %s' % (
+                ZSTORE_CLI_BIN, self.url, verb, linux.shellquote(path))
+            output = shell.call(cmdstr)
+        finally:
+            if os.path.exists(path):
+                os.remove(path)
+        try:
+            return json.loads(output)
+        except ValueError:
+            raise Exception('forest %s returned invalid JSON: %s' % (verb, output))
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)